src/scheduler/MpiWorld.cpp

```diff
@@ -176,6 +176,54 @@ std::string MpiWorld::getHostForRank(int rank)
     return rankHostMap[rank];
 }
 
+void MpiWorld::getCartesianRank(int rank, int* dims, int* periods, int* coords)
+{
+    if (rank > this->size - 1) {
+        throw std::runtime_error(
+          fmt::format("Rank {} bigger than world size {}", rank, this->size));
+    }
+
+    // Compute the corresponding rank in a 2-dim grid given the original process
+    // rank.
+    // Note - this operation, when restricted to 2dims is quivalent to:
+    // coords = {rank / sideLength, rank % sideLength}
+    int sideLength = static_cast<int>(std::floor(std::sqrt(this->size)));
+    int nprocs = sideLength * sideLength;
+    if (rank >= nprocs) {
+        for (uint8_t i = 0; i < MPI_CART_MAX_DIMENSIONS; i++) {
```

Comment:
Using an 8-bit unsigned integer as an array index and loop variable isn't wrong, but it's not in the same style as the rest of the code (which would use an int). What's the reason for this?